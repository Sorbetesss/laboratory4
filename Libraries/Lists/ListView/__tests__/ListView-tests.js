/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */
'use strict';

jest.disableAutomock();

const React = require('React');
const ReactTestRenderer = require('react-test-renderer');
const ListView = require('ListView');
const Text = require('Text');

describe('ListView', () => {
  var ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
  var datasource = ds.cloneWithRows(['John', 'Joel', 'James', 'Jimmy', 'Jackson', 'Jillian', 'Julie', 'Devin']);

  it('should render a ListView', () => {
    const component = ReactTestRenderer.create(
      <ListView
        dataSource={datasource}
        renderRow={rowData => <Text>{rowData}</Text>}
      />
    );

    expect(component).toMatchSnapshot();
  });

  xit('should be able to scroll', () => {
    // Not sure how to test this
    let ref;
    const component = ReactTestRenderer.create(
      <ListView
        ref={el => { ref = el; }}
        dataSource={datasource}
        renderRow={rowData => <Text>{rowData}</Text>}
      />
    );

    ref.scrollTo(4, 3, true);
  });
});
