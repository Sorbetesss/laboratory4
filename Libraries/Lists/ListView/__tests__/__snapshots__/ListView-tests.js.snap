// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView should render a ListView 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 8,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      John
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Joel
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      James
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Jimmy
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Jackson
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Jillian
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Julie
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Devin
    </Text>
  </View>
</RCTScrollView>
`;
